### ISSUE TITLE
error in installation process

### ISSUE THREAD

**Issue by citron96** [COMBINE-lab/salmon#497]
Hi there,

I'm a newcomer here.
Salmon source was downloaded and extracted, new folder "build" has created, 'cmake ..' has also be done to confirm the feasibility of installation.
However, When I run the code "sudo make", it returns a error regarding downloading the package tbb-2019_U8, here is the report.

```
tbb-2019_U8.tgz: FAILED
sha256sum: WARNING: 1 computed checksum did NOT match
tbb-2019_U8.tgz did not match expected SHA256! Exiting.
CMakeFiles/libtbb.dir/build.make:107: recipe for target 'libtbb-prefix/src/libtbb-stamp/libtbb-download' failed
make[2]: *** [libtbb-prefix/src/libtbb-stamp/libtbb-download] Error 1
CMakeFiles/Makefile2:253: recipe for target 'CMakeFiles/libtbb.dir/all' failed
make[1]: *** [CMakeFiles/libtbb.dir/all] Error 2
Makefile:179: recipe for target 'all' failed
make: *** [all] Error 2
```
it seems the requirement isn't met and the threshold should be modified?
appreciate any help!

**Comment by rob-p** [comment]
Hi @citron96,

  This is what is covered in https://github.com/COMBINE-lab/salmon/issues/496.  This is because an upstream package changed the SHA of their *tagged* releases (which is really not ideal).  Are you pulling from the master branch?  If you pull from develop, everything should build.  I can pull the changes that fix this into master.  Unfortunately, to avoid changing the signatures of _our_ tagged releases, I can't push this change back to older releases.

I think @nadyawilliams may also have a patch for the CMakeLists.txt file, which, perhaps, can be shared?

**Comment by nadyawilliams** [comment]
Hi @citron96,

The patch is quite simple and i have verified it and it works for salmon-1.1.0 version that i compiled.  Here is the patch content:

--- salmon-1.1.0/CMakeLists.txt.orig    2020-03-24 08:50:22.681000000 -0700
+++ salmon-1.1.0/CMakeLists.txt 2020-03-24 08:51:41.786000000 -0700
@@ -596,7 +596,7 @@
 message("Build system will fetch and build Intel Threading Building Blocks")
 message("==================================================================")
 # These are useful for the custom install step we'll do later
-set(TBB_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/tbb-2019_U8)
+set(TBB_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/oneTBB-2019_U8)
 set(TBB_INSTALL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/install)

 if("${TBB_COMPILER}" STREQUAL "gcc")
@@ -610,9 +610,9 @@
 externalproject_add(libtbb
   DOWNLOAD_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external
   DOWNLOAD_COMMAND curl -k -L https://github.com/intel/tbb/archive/2019_U8.tar.gz -o tbb-2019_U8.tgz &&
-    ${SHASUM} 7b1fd8caea14be72ae4175896510bf99c809cd7031306a1917565e6de7382fba tbb-2019_U8.tgz &&
+    ${SHASUM} 6b540118cbc79f9cbc06a35033c18156c21b84ab7b6cf56d773b168ad2b68566 tbb-2019_U8.tgz &&
     tar -xzvf tbb-2019_U8.tgz
-  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/tbb-2019_U8
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/oneTBB-2019_U8
   INSTALL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/install
   PATCH_COMMAND "${TBB_PATCH_STEP}"
   CONFIGURE_COMMAND ""


Rob, 
I understand that you don't want to push changes to older releases but perhaps one
can issue a README/NOTE for all prior releases that are affected by this. The explanation of
what changed will allow people to create their own patches for their specific releases.

Regards,
Nadya

**Comment by nadyawilliams** [comment]
Please see attached patch here.  I did not realized my writing was stylized which
wiped out correct syntax. Nadya
[salmon-1.1.0.patch.txt](https://github.com/COMBINE-lab/salmon/files/4382067/salmon-1.1.0.patch.txt)

**Comment by citron96** [comment]
get it, thank you so much. Citron




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"nadyawilliams"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月26日(星期四) 凌晨0:18
收件人:&nbsp;"COMBINE-lab/salmon"<salmon@noreply.github.com&gt;;
抄送:&nbsp;"1328515960"<1328515960@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [COMBINE-lab/salmon] error in installation process (#497)





 
Please see attached patch here.  I did not realized my writing was stylized which
 wiped out correct syntax. Nadya
 salmon-1.1.0.patch.txt

**Comment by citron96** [comment]
> Please see attached patch here. I did not realized my writing was stylized which
> wiped out correct syntax. Nadya
> [salmon-1.1.0.patch.txt](https://github.com/COMBINE-lab/salmon/files/4382067/salmon-1.1.0.patch.txt)

Hello, @nadyawilliams 
Thank you for your kind support. Since I'm pretty new here and even know nothing about how to applying this patch to fix the problem, could you give me some extra advice on it? Many thanks.

**Comment by nadyawilliams** [comment]
@citron96 
Applying patch is very standard. The way i created it was to make a copy of
salmon-1.1.0/CMakeLists.txt as salmon-1.1.0/CMakeLists.txt.orig and to make edits to
CMakeLists.txt. 
Then apply diff command to create patch:
diff -Naur salmon-1.1.0/CMakeLists.txt.orig salmon-1.1.0/CMakeLists.txt > salmon-1.1.0.patch

Now, to apply the patch one needs to put a patch file on the same level where your untar'ed salmon-1.1.0/ is and then
cd salmon-1.1.0/
patch -p1 < ../salmon-1.1.0.patch

nadya

**Comment by citron96** [comment]
> @citron96
> Applying patch is very standard. The way i created it was to make a copy of
> salmon-1.1.0/CMakeLists.txt as salmon-1.1.0/CMakeLists.txt.orig and to make edits to
> CMakeLists.txt.
> Then apply diff command to create patch:
> diff -Naur salmon-1.1.0/CMakeLists.txt.orig salmon-1.1.0/CMakeLists.txt > salmon-1.1.0.patch
> 
> Now, to apply the patch one needs to put a patch file on the same level where your untar'ed salmon-1.1.0/ is and then
> cd salmon-1.1.0/
> patch -p1 < ../salmon-1.1.0.patch
> 
> nadya

get it, and already solved the problem, thank you so much.

**Comment by citron96** [comment]
> Hi @citron96,
> 
> This is what is covered in #496. This is because an upstream package changed the SHA of their _tagged_ releases (which is really not ideal). Are you pulling from the master branch? If you pull from develop, everything should build. I can pull the changes that fix this into master. Unfortunately, to avoid changing the signatures of _our_ tagged releases, I can't push this change back to older releases.
> 
> I think @nadyawilliams may also have a patch for the CMakeLists.txt file, which, perhaps, can be shared?

thank you so much.

**Comment by nadyawilliams** [comment]
glad it was useful.

**Comment by TiongSun** [comment]
I faced the same problem and found a simple solution. The trick is at line 585 of the cMakeList.txt

"if (${TBB_VERSION} VERSION_GREATER_EQUAL 2018.0)"

It checks if you have tbb version 2018 or above. If you install tbb BEFORE running cmake, it will fulfill the requirement and bypass installing tbb in the make command, hence bypass the error.

The solution:
1. Delete the salmon folder and download a fresh one from github
2. sudo apt update     (this step is very important, to update the packages to be above version 2018) 
3. sudo apt-get install libtbb-dev
4. (Optional) apt-cache policy libtbb-dev (check the version of libtbb, it should be 2019 or above)
5. Then follows the standard installation (cmake, make etc.) The error should disappear and compile successfully

Thanks!

**Comment by rob-p** [comment]
Thanks all.  This has been fixed in 1.2.0.
